File: lib/android/src/main/java/com/rnadapty/react/models/AdaptySubscriptionPeriod.kt
package com.rnadapty.react.models

import com.adapty.models.PeriodUnit
import com.adapty.models.ProductSubscriptionPeriodModel
import com.google.gson.annotations.SerializedName

class AdaptySubscriptionPeriod(subscriptionPeriod: ProductSubscriptionPeriodModel) {
    @SerializedName("numberOfUnits")
    var numberOfUnits = subscriptionPeriod.numberOfUnits
    @SerializedName("unit")
    var unit: String = when (subscriptionPeriod.unit) {
        PeriodUnit.D -> "day"
        PeriodUnit.W -> "week"
        PeriodUnit.M -> "month"
        PeriodUnit.Y -> "year"
        null -> "unknown"
    }
}

File: lib/ios/Models/AdaptySubscriptionPeriod.swift
//
//  AdaptySubscriptionPeriod.swift
//  RNAdapty
//
//  Created by Ivan on 16.07.2021.
//

import Adapty
import Foundation

struct AdaptySubscriptionPeriod: Encodable {
  let unit: String
  let numberOfUnits: Int

  init(_ period: ProductSubscriptionPeriodModel) {
    numberOfUnits = period.numberOfUnits

    switch period.unit {
    case .day:
      unit = "day"
    case .week:
      unit = "week"
    case .month:
      unit = "month"
    case .year:
      unit = "year"
    case .unknown:
      unit = "unknown"
    }
  }
}
